tests/unit/test_tabular_data.py

```diff
@@ -252,6 +252,32 @@ def test_drop_duplicates(axis, expected):
     assert list(td) == expected
 
 
+def test_drop_duplicates_rich_text():
+    from dvc.ui import ui
+
+    td = TabularData(["col-1", "col-2", "col-3"], fill_value="-")
+
+    td.extend(
+        [
+            ["foo", None, ui.rich_text("-")],
+            ["foo", "foo"],
+            ["foo", "foo"],
+            ["foo", "bar", "foobar"],
+        ]
+    )
+
+    assert list(td) == [
```

Comment:
LGTM for the most part has only one problem here. Empty values are not the same thing as a non-empty one, should we drop them here?

Reply:
Not sure I fully understand

Reply:
Here, we drop a column if all of them share the same value, or some of them are `None` and others share the same. But the `None` value is not the same thing as a definite one they might behave differently in some cases.

Reply:
Added a flag to configure: `nan_is_value`. 

```
    assert list(td) == [
        ["foo", "-", "-"],
        ["foo", "foo", "-"],
        ["foo", "foo", "-"],
        ["foo", "bar", "foobar"],
    ]

    td.drop_duplicates("cols", nan_is_value=True)

    assert list(td) == [
        ["-", "-"],
        ["foo", "-"],
        ["foo", "-"],
        ["bar", "foobar"],
    ]
```

Defaults to `False` to preserve old behavior